src/components/image/content/Image.tsx

```diff
@@ -1,20 +1,24 @@
 import * as React from 'react';
-import { StyleImage } from '../styles/image.styles';
+import { ImageContainer, StyledImage } from '../styles/image.styles';
 import { ImageProps } from './image_types';
 
 const Image: React.FC<ImageProps> = ({ src, ...props }: ImageProps) => {
+  const [imgSrc, setImgSrc] = React.useState<string | undefined>(src);
```

Comment:
hichvaght az props mostaghim naro too state in ghalate

Reply:
i've added onError to types